### PR TITLE
Add metadata to model versions

### PR DESCRIPTION
**Pull request by avishniakov**
## Describe changes
I implemented metadata support for Model Versions.
Usage:
```python
# object-oriented outside of a pipeline
mv = ModelVersion(name=MODEL_NAME, version="latest")
mv.log_metadata({"foo": "bar"})
assert mv.metadata["foo"] == "bar" # True

# object-oriented inside a pipeline
@step(...)
def step_1():
    mv = get_step_context().model_version
    mv.log_metadata({"foo": "bar"})
    assert mv.metadata["foo"] == "bar" # True

# functional outside of a pipeline
log_model_version_metadata({"foo": "bar"}, model_name=MODEL_NAME, model_version="latest")
mv = ModelVersion(name=MODEL_NAME, version="latest")
assert mv.metadata["foo"] == "bar" # True

# functional inside a pipeline
@step(...)
def step_1():
    log_model_version_metadata({"foo": "bar"})
    mv = get_step_context().model_version
    assert mv.metadata["foo"] == "bar" # True
```

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/stacks-and-components/component-guide) table and the [corresponding website section](https://zenml.io/integrations).
- [x] I have added tests to cover my changes.
- [x] I have based my new branch on `develop` and the open PR is targeting `develop`. If your branch wasn't based on develop read [Contribution guide on rebasing branch to develop](https://github.com/zenml-io/zenml/blob/main/CONTRIBUTING.md#-pull-requests-rebase-your-branch-on-develop).

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: tests/integration/functional/model/test_model_version.py [hunk 1]
```diff
@@ -217,3 +225,67 @@ def test_tags_properly_updated(self):
                 model = mv._get_or_create_model()
                 assert len(model.tags) == 2
                 assert {t.name for t in model.tags} == {"foo", "bar"}
+
+    def test_metadata_logging(self):
```
Comment (strickvl) [inline tests/integration/functional/model/test_model_version.py:229]: Could you add testing for the failure modes of this logging?

File: tests/integration/functional/model/test_model_version.py [hunk 2]
```diff
@@ -62,6 +64,12 @@ def __exit__(self, exc_type, exc_value, exc_traceback):
             pass
 
 
+@step
+def step_metadata_logging_functional():
```
Comment (strickvl) [inline tests/integration/functional/model/test_model_version.py:68]: Sorry for the nit, but makes it a bit easier to navigate sometimes... Could you add a docstring explaining what this tests?

**Review by strickvl — COMMENTED** [review]
LGTM, though wondering again whether this needs a DB migration...

File: src/zenml/models/model_models.py [hunk 3]
```diff
@@ -158,6 +161,7 @@ class ModelVersionResponseModel(
         description="Pipeline runs linked to the model version",
         default={},
     )
+    run_metadata: Dict[str, "RunMetadataResponse"] = {}
```
Comment (fa9r) [inline src/zenml/models/model_models.py:164]: Maybe turn into a field with description etc.

**Review by fa9r — APPROVED** [review]
Looks good to go 🙌

File: tests/integration/functional/model/test_model_version.py (hunk 1 above)
Comment (avishniakov) [reply tests/integration/functional/model/test_model_version.py:229]: I have it in `test_metadata_logging_functional`, it cannot fail in the object-oriented way on metadata logging (unless unexpected issues), only failing before the logging is possible, e.g. cannot get ModelVersion or else.

File: tests/integration/functional/model/test_model_version.py (hunk 2 above)
Comment (avishniakov) [reply tests/integration/functional/model/test_model_version.py:64]: Fixed

**Pull request merged**